packages/ui/src/components/SearchModal/SearchModal.tsx

```diff
@@ -0,0 +1,76 @@
+'use client';
+
+import type { Category, Culture, UploadedImage } from '@prezly/sdk';
+import * as Dialog from '@radix-ui/react-dialog';
+import algoliasearch from 'algoliasearch/lite';
+import { useMemo } from 'react';
+import { Configure, InstantSearch } from 'react-instantsearch';
+import { twMerge } from 'tailwind-merge';
+
+import { MainPanel, SearchBar } from './components';
+import type { AlgoliaConfig } from './types';
+
+export interface Props {
+    isOpen: boolean;
+    className?: string;
+    overlayClassName?: string;
+    onOpenChange: (open: boolean) => void;
+    algoliaConfig: AlgoliaConfig;
+    locale: Culture['code'];
+    newsroomName: string;
+    categories: Category[];
+    hideSubtitle: boolean;
+    showDate: boolean;
+    logo: UploadedImage | null;
+}
+
+export function SearchModal({
+    isOpen,
+    categories,
+    className,
+    overlayClassName,
+    onOpenChange,
+    algoliaConfig,
+    locale,
+    newsroomName,
+    logo,
+    hideSubtitle,
+    showDate,
+}: Props) {
+    const { ALGOLIA_API_KEY, ALGOLIA_APP_ID, ALGOLIA_INDEX } = algoliaConfig;
+
+    const searchClient = useMemo(
+        () => algoliasearch(ALGOLIA_APP_ID, ALGOLIA_API_KEY),
+        [ALGOLIA_API_KEY, ALGOLIA_APP_ID],
+    );
+
+    return (
+        <Dialog.Root open={isOpen} onOpenChange={onOpenChange}>
```

Comment:
❌ 

The modal can't be closed on mobile 😕

Reply:
https://github.com/prezly/theme-kit-js/pull/440/commits/99f78ceb1773d56de7efe17a3fc63ce2a45a4001